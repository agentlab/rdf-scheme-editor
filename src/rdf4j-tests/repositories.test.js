import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';
import RepositoriesTable from '../components/Repositories.stories';

const server_URL = 'https://agentlab.ru/rdf4j-server';
const repositories_prefix = '/repositories';
const props = ['id', 'title', 'uri', 'readable', 'writable'];

let webData = fetch(server_URL.concat(repositories_prefix), {
  method: 'GET',
  headers: {
    Accept: 'application/sparql-results+json',
  },
});

let wrongWebData = fetch(server_URL.concat(repositories_prefix).concat('WRONG_POSTFIX'), {
  method: 'GET',
  headers: {
    Accept: 'application/sparql-results+json',
  },
});

test('Renders correctly if data is correct', () => {
  configure({ adapter: new Adapter() });
  const output = shallow(<RepositoriesTable props={server_URL.concat(repositories_prefix)} />);
  expect(shallowToJson(output)).toMatchSnapshot({
    dataSource: expect.any(Array),
  });
});

test('Render empty table if no data', () => {
  configure({ adapter: new Adapter() });
  const output = shallow(<RepositoriesTable props={server_URL.concat(repositories_prefix).concat('WRONG_POSTFIX')} />);
  expect(shallowToJson(output)).toMatchSnapshot({
    // empty table data
  });
});

test('Web request is made correctly', async () => {
  await webData.then((r) => {
    expect(r.status).toEqual(200);
  });
});

test('Web request to a wrong address is caught', async () => {
  await wrongWebData.then((r) => {
    expect(r.status).not.toEqual(200);
  });
});

test('The repositories information is correct', async () => {
  const data = await webData.then((r) => r.json());

  expect(data).not.toBeUndefined();

  data.results.bindings.forEach((dataElement) => {
    props.forEach((property) => {
      expect(dataElement).toHaveProperty(property);
    });
  });
});
