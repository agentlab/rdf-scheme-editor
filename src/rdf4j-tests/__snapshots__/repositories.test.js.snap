// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders corectly 1`] = `
<Table
  bordered={true}
  className=""
  dataSource={Array []}
  indentSize={20}
  loading={false}
  locale={Object {}}
  pagination={false}
  rowKey="key"
  showHeader={true}
  size="small"
  sortDirections={
    Array [
      "ascend",
      "descend",
    ]
  }
  useFixedHeader={false}
/>
`;
